packages/shared/lib/core/wallet/types/new-transaction-details.type.ts

```diff
@@ -0,0 +1,33 @@
+import type { FeatureTypes } from '@iota/types'
+import { IAsset, Subject, NewTransactionType } from '@core/wallet'
+
+export type NewTransactionDetails = NewNftTransactionDetails | NewTokenTransactionDetails
+
+export type NewTokenTransactionDetails = {
+    type: NewTransactionType.TokenTransfer
+    rawAmount: string
+    asset: IAsset
+    unit: string
+    metadata: string
+    recipient: Subject
+    tag: string
+    expirationDate?: Date
+    giftStorageDeposit?: boolean
+    surplus?: string
+    disableToggleGift?: boolean
+    disableChangeExpiration?: boolean
```

Comment:
Maybe have a common TransactionDetails type that is extended in this file to avoid duplication?